juicer/scikit_learn/etl_operation.py

```diff
@@ -15,15 +15,15 @@ class AddColumnsOperation(Operation):
     def __init__(self, parameters, named_inputs, named_outputs):
         Operation.__init__(self, parameters,  named_inputs, named_outputs)
 
-        self.has_code = len(named_inputs) == 2
+        self.has_code = len(named_outputs) == 1 and len(named_inputs) == 2
 
         self.suffixes = parameters.get(self.ALIASES_PARAM, '_ds0,_ds1')
         self.suffixes = [s for s in self.suffixes.replace(" ", "").split(',')]
 
         if not self.has_code:
```

Comment:
I know you didn't add this validation, you may remove it. Even though there is no code, the operation can be added to a workflow. In general, check only the cases when required parameters are empty.